include/IECore/IndexedIOAlgo.h

```diff
@@ -0,0 +1,92 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECore/Export.h"
+
+#include "IECore/IndexedIO.h"
+
+namespace IECore
+{
+	namespace IndexedIOAlgo
+	{
+		template<typename T>
+		struct FileStats
```

Comment:
I had to read the implementation of this before I could make sense of what I might find in `numBytes` and `numBlocks`. Couple of thoughts that may or may not help :

- We could move the "histogram of blockSize and byte totals for 64 power 2 bins" comment onto the FileStats class itself.
- Renaming `numBits` to `numBins` might make it clearer that the arrays are bins.
- Providing a noddy `binBlockSize( binIndex )` method might drive the point home even more
- The templating confused me, because I didn't see why any other type would be appropriate - that only became apparent when looking at the internal implementation of `copy()`.

Mostly though, I think a nicely formatted ostream operator might be the most useful thing we could add. I suspect that's the main thing I'd want - an easy way of dumping the results into a stats file.

Reply:
I don't think my comments here are as important as my comments about the overloading of the API. If we only have time to address some things, then I would address the overloading comments.

Reply:
I did originally have a C++ stream operator (but deleted my cortex done) but thought in most cases you'd dump the output in python. I'll add the  ostream<< operator also.